bot/exts/utils/utils.py

```diff
@@ -88,40 +88,53 @@ def get_info(char: str) -> tuple[str, str]:
     async def zen(
         self,
         ctx: Context,
-        zen_rule_index: int | None,
-        *,
-        search_value: str | None = None
+        search_value: str | None,
     ) -> None:
         """
         Show the Zen of Python.
 
         Without any arguments, the full Zen will be produced.
-        If zen_rule_index is provided, the line with that index will be produced.
-        If only a string is provided, the line which matches best will be produced.
+        If an index or a slice is provided, the corresponding lines will be produced.
+        Otherwise, the line which matches best will be produced.
         """
         embed = Embed(
             colour=Colour.og_blurple(),
             title="The Zen of Python",
             description=ZEN_OF_PYTHON
         )
 
-        if zen_rule_index is None and search_value is None:
+        if search_value is None:
             embed.title += ", by Tim Peters"
             await ctx.send(embed=embed)
             return
 
         zen_lines = ZEN_OF_PYTHON.splitlines()
 
-        # Prioritize passing the zen rule index
-        if zen_rule_index is not None:
-
+        # Prioritize checking for an index or slice
+        match = re.search(r"^(\d+)(:\d+)?", search_value.split(" ")[0])
+        if match:
```

Comment:
To regex needs to be adjusted to account for negative indices:
```suggestion
        match = re.search(r"^(-?\d+)(:-?\d+)?", search_value.split(" ")[0])
        if match:
```
(Rest of the logic already handles negative indexing fine)

Reply:
With the current regex, the match fails and `search_value` is assumed to be a string query rather than as an index or a slice.

Reply:
I'll add a small nitpick: `re.search(r"^PATTERN")` can be written as `re.match(r"PATTERN")`

Reply:
right, I thought about negative indices everywhere but in the regex 😅. I'll add these changes later on today

Reply:
Maybe it's possible to write a test for this? It seems like a mostly-pure function